### ISSUE TITLE
Allow customisation of optimisation defaults

### ISSUE THREAD

**Issue by ocaisa** [hpcugent/easybuild-framework#1414]
The current default compiler optimisation is `-O2` but optimisation defaults should really be configurable, ideally in the toolchain easyconfig.

**Comment by boegel** [comment]
Some ideas:

* add support for `default_toolchainopts` that can be defined in the toolchain easyconfig (not sure how those would be picked up when the toolchain is used though, since the toolchain easyconfig is typically out of the picture then if the corresponding module is already available)
* add support for `--optflags` to override whatever optimisation flags would be used (regardless of whether they're default or steered by the easyconfig)

@ocaisa: you should already be able to tweak the defaults on a per-toolchain basis by customising the toolchain definition (and make sure it's used via `--include-toolchains`), something like (in `ps:

```python
self.COMPILER_SHARED_OPTS['opt'][0] = True
```

In the `.py` module for whichever toolchain you're using (in the right place, of course ;-)).

**Comment by ocaisa** [comment]
What about setting an env var that can be pulled in from the module file of the toolchain, say like `EASYBUILD_CUSTOMOPTS` which we parse to set the appropriate values? Overriding these would just mean  setting a new value on the command line?

**Comment by ocaisa** [comment]
Your suggestion doesn't work because the value was immutable, I had to use
```
from easybuild.toolchains.compiler.gcc import Gcc

class GNU(Gcc):
    """Compiler-only toolchain, including only GCC and binutils."""
    NAME = 'GNU'
    # Increase the default optimisation
    Gcc.COMPILER_SHARED_OPTS['opt'] = (True, "High compiler optimizations")
```

**Comment by boegel** [comment]
@ocaisa: fixed with #1565?

**Comment by ocaisa** [comment]
Yes, I'm happy with the solution there
